### PR TITLE
Update Ciphers Support

### PR DESCRIPTION
**Pull request by Jason-Morcos**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Just a stab at changing the ciphers list to the list found by @gboudreau in issue #164 
Probably needs some testing, tweaks, and forethought here - for instance, which ciphers should normally be disabled?

Does this close any currently open issues?
------------------------------------------
Aimed at #164 and #419 

Any relevant logs, error output, etc?
-------------------------------------
Nada

Any other comments?
-------------------
Someone else can pick this up here and make a real final PR with changes

Where has this been tested?
---------------------------
**Devices/Simulators:** …

**macOS Version:** …

**Sequel-Ace Version:** …

### CODE REVIEW

**Comment by robinkunde** [comment]
2 useful links:
https://www.cisco.com/c/en/us/support/docs/security/email-security-appliance/210533-SSL-Cipher-Strength-Details.html
https://testssl.sh/openssl-iana.mapping.html

IMO, every cipher that's use on SSLv3 should probably be disabled by default. The protocol is insecure and Chrome for example disabled it by default quite a while ago in favor of TLS.

**Comment by Kaspik** [comment]
On the other hand you need to keep them in app because of older MySQL versions, right? MySQL 8 has different list of supported Ciphers than MySQL 5.6.

The best would be to find a way how to get list of supported Ciphers from the MySQL itself... But I have zero knowledge here, so I'll leave this up to anyone else :)

**Comment by Kaspik** [comment]
BTW, I like this comment: https://github.com/Sequel-Ace/Sequel-Ace/issues/164#issuecomment-685817445

`You could even switch from a predefined list to a text input the user has to fill himself. The one who has special needs will know which ciphers to put in. This would eliminate all work updating this list in the future without restraining power users.`
For 2.x.x I would just update the list by adding new, for 3.0.0 we could change it to TextField and make it as a breaking change - use needs to select the Cipher by filling it in the TextField.

**Comment by Jason-Morcos** [comment]
> BTW, I like this comment: [#164 (comment)](https://github.com/Sequel-Ace/Sequel-Ace/issues/164#issuecomment-685817445)
> 
> `You could even switch from a predefined list to a text input the user has to fill himself. The one who has special needs will know which ciphers to put in. This would eliminate all work updating this list in the future without restraining power users.`
> For 2.x.x I would just update the list by adding new, for 3.0.0 we could change it to TextField and make it as a breaking change - user needs to select the Cipher by filling it in the TextField.

Perhaps, as long as a normal user has to do _absolutely no work with ciphers_ to connect. I think we need to focus on Sequel Ace being easy for novices. A lot of our users probably aren't quite as advanced and just want it to work without any special setup.

**Comment by Jason-Morcos** [comment]
> 2 useful links:
> https://www.cisco.com/c/en/us/support/docs/security/email-security-appliance/210533-SSL-Cipher-Strength-Details.html
> https://testssl.sh/openssl-iana.mapping.html
> 
> IMO, every cipher that's use on SSLv3 should probably be disabled by default. The protocol is insecure and Chrome for example disabled it by default quite a while ago in favor of TLS.

@robinkunde Good idea, do you think you'd be able to apply these changes as a tweak to this PR? Then perhaps we would be ready to rumble with putting this into a beta at least to see if it fixes all of the cipher issues?

**Comment by robinkunde** [comment]
I did more reading and it's **possible** we can remove this feature entirely in a future update. Starting with [v5.7.10](https://dev.mysql.com/doc/refman/5.7/en/encrypted-connection-protocols-ciphers.html), MySQL actually explicitly disallows whole categories of insecure ciphers. This should apply to the client as well, and we're using v8 there now. If that's the case, we might be able to defer to the client completely and remove the list from the app.

**Comment by robinkunde** [comment]
I'll make the change to the default list for now though.

**Comment by Jason-Morcos** [comment]
@robinkunde This may be a good one to include in the 2.3.0 release so if you're able to tweak the order or want to provide me with a specific order to tweak to before the 2.3.0 release that may be good! Worst case it becomes part of 3.0, but we probably want 2.x to have a long lifetime of legacy support

**Comment by Jason-Morcos** [comment]
Actually, better question, is there a reason to _not_ just merge this PR as-is and tweak only if necessary after? @Kaspik?

File: Source/Controllers/Preferences/Panes/SPNetworkPreferencePane.m
Comment (Kaspik) [inline Source/Controllers/Preferences/Panes/SPNetworkPreferencePane.m:579]: Can we just comment this out? I don't know how the marker works, but seems "safer" to not include them at all

Comment (Jason-Morcos) [reply Source/Controllers/Preferences/Panes/SPNetworkPreferencePane.m:579]: Hmmm we could comment them out, but as long as the marker works as intended (which testing in #164 indicates it does) there's no inherent danger in including less secure ciphers as long as they are disabled by default. I worry that if we remove these less secure ciphers, some users won't be able to connect to their legacy databases which may not support the more secure ciphers at all. This way, they would have to go in and manually allow a less secure cipher, but we're not making the app unusable for these users.

**Comment by Jason-Morcos** [comment]
Updated PR based on feedback from @Synchro
https://github.com/Sequel-Ace/Sequel-Ace/issues/164#issuecomment-718757500

**Comment by Jason-Morcos** [comment]
@stychos If we do make this change, might be a good idea to add a section to the FAQ about "why can't I connect" or something and mention ciphers? Not sure if that would be helpful, but seems we may get some questions from this

**Review by Kaspik — APPROVED** [review]

**Pull request merged**
